olp-cpp-sdk-core/tests/logging/MessageFormatterTest.cpp

```diff
@@ -0,0 +1,501 @@
+    EXPECT_EQ( "%lu", threadIdElement.format );
+    EXPECT_EQ( 0, threadIdElement.limit );
+
+    MessageFormatter::Element arbitraryElement( MessageFormatter::ElementType::File, "%30s", -30 );
+    EXPECT_EQ( MessageFormatter::ElementType::File, arbitraryElement.type );
+    EXPECT_EQ( "%30s", arbitraryElement.format );
+    EXPECT_EQ( -30, arbitraryElement.limit );
+}
+
+TEST( MessageFormatterTest, ElementMove )
+{
+    MessageFormatter::Element element( MessageFormatter::ElementType::File, "%30s", -30 );
+
+    MessageFormatter::Element moveConstructed( std::move( element ) );
+    EXPECT_EQ( MessageFormatter::ElementType::File, moveConstructed.type );
+    EXPECT_EQ( "%30s", moveConstructed.format );
+    EXPECT_EQ( -30, moveConstructed.limit );
+    EXPECT_TRUE( element.format.empty( ) );
+
+    MessageFormatter::Element moveAssigned( MessageFormatter::ElementType::String );
+    moveAssigned = std::move( moveConstructed );
+    EXPECT_EQ( MessageFormatter::ElementType::File, moveAssigned.type );
+    EXPECT_EQ( "%30s", moveAssigned.format );
+    EXPECT_EQ( -30, moveAssigned.limit );
+    EXPECT_TRUE( moveConstructed.format.empty( ) );
+}
+
+TEST( MessageFormatterTest, ElementEquality )
+{
+    MessageFormatter::Element element1( MessageFormatter::ElementType::File, "%30s", -30 );
+    MessageFormatter::Element element2( element1 );
+
+    EXPECT_TRUE( element1 == element2 );
+    EXPECT_FALSE( element1 != element2 );
+
+    element2.type = MessageFormatter::ElementType::Line;
+    EXPECT_FALSE( element1 == element2 );
+    EXPECT_TRUE( element1 != element2 );
+
+    element2 = element1;
+    element2.format = "%u";
+    EXPECT_FALSE( element1 == element2 );
+    EXPECT_TRUE( element1 != element2 );
+
+    element2 = element1;
+    element2.limit = 20;
+    ;
+    EXPECT_FALSE( element1 == element2 );
+    EXPECT_TRUE( element1 != element2 );
+}
+
+TEST( MessageFormatterTest, DefaultLevelMap )
+{
+    const MessageFormatter::LevelNameMap& map = MessageFormatter::defaultLevelNameMap( );
+    EXPECT_EQ( "[TRACE]", map[ static_cast< std::size_t >( Level::Trace ) ] );
+    EXPECT_EQ( "[DEBUG]", map[ static_cast< std::size_t >( Level::Debug ) ] );
+    EXPECT_EQ( "[INFO]", map[ static_cast< std::size_t >( Level::Info ) ] );
+    EXPECT_EQ( "[WARN]", map[ static_cast< std::size_t >( Level::Warning ) ] );
+    EXPECT_EQ( "[ERROR]", map[ static_cast< std::size_t >( Level::Error ) ] );
+    EXPECT_EQ( "[FATAL]", map[ static_cast< std::size_t >( Level::Fatal ) ] );
+}
+
+TEST( MessageFormatterTest, DefaultConstructor )
+{
+    MessageFormatter formatter;
+    EXPECT_TRUE( formatter.getElements( ).empty( ) );
+    EXPECT_EQ( MessageFormatter::defaultLevelNameMap( ), formatter.getLevelNameMap( ) );
+    EXPECT_EQ( MessageFormatter::Timezone::Local, formatter.getTimezone( ) );
+}
+
+TEST( MessageFormatterTest, Constructor )
+{
+    std::vector< MessageFormatter::Element > elements
```

Comment:
```suggestion
static const  std::vector< MessageFormatter::Element > elements
```
No need to construct const vector each time test is run

Reply:
I don't mind changing but don't think that it's something to worry about. Such optimization gets a fraction of extra performance. And not to mention that compilers will probably optimize that anyway.